src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -761,33 +772,28 @@ namespace OpenMS
 
       for (IsotopeDistribution::ConstIterator it = dist.begin(); it != dist.end(); ++it)
       {
-        p.setMZ(it->getMZ() / charge);
-        p.setIntensity(pre_int_H2O_ *  it->getIntensity());
         if (add_metainfo_)
         {
-          String ion_name("[M+H]-H2O" + String((Size)abs(charge), '+'));
-          ion_names.push_back(ion_name);
+          ion_names.push_back(ion_name_h2o);
```

Comment:
std::move as ion_name_doesn't get used again?

Reply:
Good idea, don't know if it will make a difference, as SSO will be used for the String (it's only 9+charge long).